### ISSUE TITLE
helm deployment: core-command crash when deployed on Baetyl framework

### ISSUE THREAD

**Issue by zhangtag** [edgexfoundry/edgex-examples#108]
i try to deploy edgex2.0 on k8s , but the core-command crashed
```
`level=INFO ts=2022-05-03T03:37:37.920540664Z app=core-command source=variables.go:352 msg="Variables override of 'Startup Duration' by environment variable: EDGEX_STARTUP_DURATION=120"  
level=INFO ts=2022-05-03T03:37:37.921450878Z app=core-command source=config.go:359 msg="Loaded service configuration from /res/configuration.toml"  
level=INFO ts=2022-05-03T03:37:37.922112252Z app=core-command source=variables.go:352 msg="Variables override of 'Clients.core-metadata.Host' by environment variable: CLIENTS_CORE_METADATA_HOST=edgex-core-metadata"
level=INFO ts=2022-05-03T03:37:37.922187853Z app=core-command source=variables.go:352 msg="Variables override of 'Databases.Primary.Host' by environment variable: DATABASES_PRIMARY_HOST=edgex-redis"
level=INFO ts=2022-05-03T03:37:37.922236097Z app=core-command source=variables.go:352 msg="Variables override of 'Registry.Host' by environment variable: REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-05-03T03:37:37.922248694Z app=core-command source=variables.go:352 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-core-command"
level=INFO ts=2022-05-03T03:37:37.922371208Z app=core-command source=config.go:156 msg="Using Config Provider access token of length 0"
level=INFO ts=2022-05-03T03:37:37.922431493Z app=core-command source=config.go:334 msg="Using Configuration provider (consul) from: http://edgex-core-consul:8500 with base path of edgex/core/2.0/core-command"
level=INFO ts=2022-05-03T03:37:37.973392331Z app=core-command source=variables.go:352 msg="Variables override of 'Clients.core-metadata.Host' by environment variable: CLIENTS_CORE_METADATA_HOST=edgex-core-metadata"
level=INFO ts=2022-05-03T03:37:37.973527561Z app=core-command source=variables.go:352 msg="Variables override of 'Databases.Primary.Host' by environment variable: DATABASES_PRIMARY_HOST=edgex-redis"
level=INFO ts=2022-05-03T03:37:37.973555234Z app=core-command source=variables.go:352 msg="Variables override of 'Registry.Host' by environment variable: REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-05-03T03:37:37.973584251Z app=core-command source=variables.go:352 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-core-command"
level=INFO ts=2022-05-03T03:37:37.973695764Z app=core-command source=config.go:494 msg="Configuration has been pulled from Configuration provider (4 envVars overrides applied)"
level=INFO ts=2022-05-03T03:37:37.973758219Z app=core-command source=registry.go:57 msg="Using Registry access token of length 0"
level=INFO ts=2022-05-03T03:37:37.973780185Z app=core-command source=registry.go:73 msg="Using Registry (consul) from http://edgex-core-consul:8500"
level=INFO ts=2022-05-03T03:37:37.987505155Z app=core-command source=telemetry.go:86 msg="Telemetry starting"
level=INFO ts=2022-05-03T03:37:37.987579929Z app=core-command source=httpserver.go:114 msg="Web server starting (edgex-core-command:59882)"
level=INFO ts=2022-05-03T03:37:37.987598358Z app=core-command source=message.go:50 msg="Service dependencies resolved..."
level=INFO ts=2022-05-03T03:37:37.987610149Z app=core-command source=message.go:51 msg="Starting core-command 2.0.0 "
level=INFO ts=2022-05-03T03:37:37.987619381Z app=core-command source=message.go:55 msg="This is the Core Command Microservice"
level=INFO ts=2022-05-03T03:37:37.987637468Z app=core-command source=message.go:58 msg="Service started in: 66.973179ms"
level=ERROR ts=2022-05-03T03:37:37.989088642Z app=core-command source=httpserver.go:126 msg="Web server failed: listen tcp 10.98.134.67:59882: bind: cannot assign requested address"
level=INFO ts=2022-05-03T03:37:37.989230903Z app=core-command source=httpserver.go:109 msg="Web server shutting down"
level=INFO ts=2022-05-03T03:37:37.989318622Z app=core-command source=httpserver.go:111 msg="Web server shut down"
level=INFO ts=2022-05-03T03:37:38.988982755Z app=core-command source=telemetry.go:98 msg="Telemetry stopped"
level=INFO ts=2022-05-03T03:37:38.989072256Z app=core-command source=bootstrap.go:135 msg="Un-Registering service from the Registry"`
```

i have beening trapped in this for a long time , 

hope u can help me ,thanks !

**Comment by bnevis-i** [comment]
@zhangtag are you using the example helm chart in this repository, or something else?

This line suggests there might be an instance of core-command already running.

Is your kubernetes cluster working otherwise?

Can you provide more detailed instructions to reproduce?

level=ERROR ts=2022-05-03T03:37:37.989088642Z app=core-command source=httpserver.go:126 msg="Web server failed: listen tcp 10.98.134.67:59882: bind: cannot assign requested address"

**Comment by zhangtag** [comment]
> @zhangtag are you using the example helm chart in this repository, or something else?
> 
> This line suggests there might be an instance of core-command already running.
> 
> level=ERROR ts=2022-05-03T03:37:37.989088642Z app=core-command source=httpserver.go:126 msg="Web server failed: listen tcp 10.98.134.67:59882: bind: cannot assign requested address"
> 
> Is your kubernetes cluster working otherwise?
> 
> Can you provide more detailed instructions to reproduce?

In an edge computing project baetyl, the edgex microservice issued by baetyl cannot be started except for redis, consul, and ui, all of which are the errors mentioned above.

**Comment by bnevis-i** [comment]
> In an edge computing project baetyl, the edgex microservice issued by baetyl cannot be started except for redis, consul, and ui, all of which are the errors mentioned above.

For compatibility with EdgeX on docker, the default settings for the helm chart map host ports to the container.  Does Baetyl allow that?

**Comment by zhangtag** [comment]
> > In an edge computing project baetyl, the edgex microservice issued by baetyl cannot be started except for redis, consul, and ui, all of which are the errors mentioned above.
> 
> For compatibility with EdgeX on docker, the default settings for the helm chart map host ports to the container. Does Baetyl allow that?

Thanks for your reply, baetyl is an edge computing platform like kubeedge
But I still have the same problem when I deploy k8s and edgex with a completely pure new system. I guess it is a problem with a certain setting of my network environment.
`level=INFO ts=2022-05-11T03:21:47.026891864Z app=core-command source=config.go:373 msg="Loaded service configuration from /res/configuration.toml"
level=INFO ts=2022-05-11T03:21:47.027670329Z app=core-command source=variables.go:352 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-core-command"
level=INFO ts=2022-05-11T03:21:47.027731047Z app=core-command source=variables.go:352 msg="Variables override of 'Registry.Host' by environment variable: REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-05-11T03:21:47.027766072Z app=core-command source=variables.go:352 msg="Variables override of 'Clients.core-metadata.Host' by environment variable: CLIENTS_CORE_METADATA_HOST=edgex-core-metadata"
level=INFO ts=2022-05-11T03:21:47.027899043Z app=core-command source=config.go:159 msg="Using Configuration Provider access token of length 0"
level=INFO ts=2022-05-11T03:21:47.027944438Z app=core-command source=config.go:348 msg="Using Configuration provider (consul) from: http://edgex-core-consul:8500 with base path of edgex/core/2.0/core-command"
level=ERROR ts=2022-05-11T03:21:57.033542472Z app=core-command source=config.go:178 msg="configuration provider is not available"
level=ERROR ts=2022-05-11T03:22:08.040552813Z app=core-command source=config.go:178 msg="configuration provider is not available"
level=ERROR ts=2022-05-11T03:22:19.047049012Z app=core-command source=config.go:178 msg="configuration provider is not available"
level=ERROR ts=2022-05-11T03:22:30.053220564Z app=core-command source=config.go:178 msg="configuration provider is not available"
level=ERROR ts=2022-05-11T03:22:41.059138485Z app=core-command source=config.go:178 msg="configuration provider is not available"
level=ERROR ts=2022-05-11T03:22:52.061870422Z app=core-command source=config.go:178 msg="configuration provider is not available"
`

**Comment by bnevis-i** [comment]
Ok.  Planning to do some maintenance on the helm chart this summer.  Will keep this in mind.  Obviously, if you can get to the bottom of it sooner, feel free to submit a PR to tweak the sample.

**Comment by zhangtag** [comment]
> Ok. Planning to do some maintenance on the helm chart this summer. Will keep this in mind. Obviously, if you can get to the bottom of it sooner, feel free to submit a PR to tweak the sample.

No problem, thanks for your reply, I'll look into it again

**Comment by bnevis-i** [comment]
@zhangtag 

Today we closed #92 which adds helm chart support for hostPath, openebs, and rook/ceph volumes.  It should also work for NFS-based storage classes as well.

Can you please re-test with the newest release and report if we can close this issue.

Will close this issue next week if I don't hear from you.

**Comment by bnevis-i** [comment]
Closing due to inactivity.
